### ISSUE TITLE
Different results using same code but different CPU type

### ISSUE THREAD

**Issue by rluethy** [tensorflow/tensorflow#53277]
**System information**
- Linux Ubuntu 20.04
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.5.0
- Python version: 3.8

**Describe the current behavior**

When tuning TF models we noted that we were getting different results when using different instance types on AWS. We started removing possible sources of variability until the only difference was the CPU type of the instances. We created a minimal example that trains a simple MLP on a training set and predicts probabilities for a validation set. The following figure compares the predictions obtained with an Intel CPU to those with an AMD CPU:
 
![image](https://user-images.githubusercontent.com/3641872/144308840-b0415669-dcbb-4200-bb55-9815843c3e50.png)

**Describe the expected behavior**

The predicted probabilities should be the same with both CPU types and the points in the above plot should all be on the diagonal.

**Standalone code to reproduce the issue**

Code to reproduce issue: https://github.com/rluethy/tf-cpu-type-diff

**Comment by mohantym** [comment]
Hi @rluethy! Did you check with latest version 2.7 too?
Please post this issue on [keras-team/keras repo](https://github.com/keras-team/keras/issues) if it is not getting resolved.
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999) . Thanks!

**Comment by mohantym** [comment]
Ok @rluethy ! Please close this issue here as it will be tracked there.

**Comment by rluethy** [comment]
Reposted here https://github.com/keras-team/keras/issues/15740
